.teamcity/scheduled/build/base/TeamCityScheduledImageBuildLinux_Base.kt

```diff
@@ -9,97 +9,97 @@ import jetbrains.buildServer.configs.kotlin.v2019_2.BuildType
 import jetbrains.buildServer.configs.kotlin.v2019_2.buildFeatures.dockerSupport
 import jetbrains.buildServer.configs.kotlin.v2019_2.buildSteps.script
 import utils.config.DeliveryConfig
+import utils.config.Registries
+import utils.dsl.general.teamCityImageBuildFeatures
 import java.lang.IllegalArgumentException
 
 /**
  * Base class for the build of Linux-based Docker images.
  * @param platform target platform, e.g. amd64, aarch64 (ARM)
  * @param agentReq a substring that agent name ('teamcity.agent.name') must contain
  */
-class TeamCityScheduledImageBuildLinux_Base(private val platform: String, private val agentReq: String) :
-    BuildType({
-        name = "[${platform}] TeamCity Docker Images - Automated Scheduled Build - Linux"
-        id("TeamCityScheduledImageBuildLinux_Base${platform}")
+class TeamCityScheduledImageBuildLinux_Base(private val platform: String, private val agentReq: String) : BuildType({
+    name = "[${platform}] TeamCity Docker Images - Automated Scheduled Build - Linux"
+    id("TeamCityScheduledImageBuildLinux_Base${platform}")
 
+    vcs {
+        root(TeamCityDockerImagesRepo.TeamCityDockerImagesRepo)
+    }
+
+    // all .yml files (e.g. compose samples)
+    artifactRules = "+:*.yml"
+
+    params {
+        // the images will be published into registry that holds nightly builds
+        param("docker.buildRepository", "%docker.nightlyRepository%")
+        // no postfix needed
+        param("docker.buildImagePostfix", "")
+        param("tc.image.version", "%dockerImage.teamcity.buildNumber%")
+    }
+
+    steps {
         val images: Set<ImageInfo> = when {
             platform.lowercase().contains("arm") || platform.lowercase()
                 .contains("aarch") -> ImageInfoRepository.getArmLinuxImages2004(
-                "%docker.nightlyRepository%",
-                "%dockerImage.teamcity.buildNumber%"
+                stagingRepo = "%docker.nightlyRepository%",
+                version = "%dockerImage.teamcity.buildNumber%",
+                prodRepo = "%docker.nightlyRepository%",
```

Comment:
Do we need a prod repo in the daily running builds?

Reply:
@burnasheva we don't use `prodRepo` in context of this configuration, however, I've decided to set it explicitly as `nightlyRepository` in order to prevent an unwanted push into an actual production registry from other maintainers.